code/game/machinery/doors/door.dm

```diff
@@ -77,6 +78,8 @@
 	if(spark_system)
 		qdel(spark_system)
 		spark_system = null
+	if(Lock)
+		qdel(Lock)
 	return ..()
 
 /obj/machinery/door/CollidedWith(atom/movable/AM)
```

Comment:
should still check (locked) regardless, even if it's renamed to bolted or something in the future for more consistent behavior.